yjit/src/asm/mod.rs

```diff
@@ -0,0 +1,51 @@
+mod libc;
+mod memreg;
+pub mod x86_64;
+
+/// A trait for defining the requisite functions to support assembly.
+pub trait Assembler<Opnd> {
+    /// Add two operands together.
+    fn add(&mut self, opnd0: Opnd, opnd1: Opnd) -> ();
+
+    /// Bitwise and the two operands together.
+    fn and(&mut self, opnd0: Opnd, opnd1: Opnd) -> ();
+
+    /// Indirect call with an R/M operand.
+    fn call(&mut self, opnd: Opnd) -> ();
+
+    /// Compare two operands and set flags.
+    fn cmp(&mut self, opnd0: Opnd, opnd1: Opnd) -> ();
```

Comment:
Hmmm. In terms of a shared Assembler trait, I think it makes sense to have low-level operations to write bytes to a memory region, but the signatures you have here are instructions that are specific to x86. This can't be shared between x86 and arm64.